modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java

```diff
@@ -265,17 +267,18 @@ public static void addImageLoaderFactory(ImageLoaderFactory factory) {
             } else {
                 loader = getLoaderBySignature(input, listener);
             }
+            if (loader != null) {
+                images = loadAll(loader, width, height, preserveAspectRatio, pixelScale, smooth);
+            } else {
+                throw new ImageStorageException("No loader for image data");
```

Comment:
Now that this is moved inside a try/catch, this `ImageStorageException` will get wrapped in another `ImageStorageException` if it is ever thrown. You probably want to catch `ImageStorageException` below and re-throw it without wrapping it.

Reply:
I will change the catch as below and update with next commit.
```
} catch (ImageStorageException ise) {
        throw ise;
} catch (IOException e) {
        throw new ImageStorageException(e.getMessage(), e);
} 
```

Reply:
Updated the catch statement.